integration_tests/suite/test_cdr_recording.py

```diff
@@ -148,3 +150,140 @@ def test_get_media_using_token_tenant_query_string(self):
         assert_that(response.status_code, equal_to(404))
+
+    # Deleting
+
+    @call_log(
+        **{'id': 1},
+        date='2021-01-01T01:00:00+01:00',
+        recordings=[{'path': '/tmp/foobar.wav'}, {'path': '/tmp/foobar2.wav'}],
+    )
+    def test_delete_media(self):
+        cdr_id = 1
+        self.filesystem.create_file('/tmp/foobar.wav', content='deleted')
+        self.filesystem.create_file('/tmp/foobar2.wav', content='visible')
+        recording1 = self.call_logd.cdr.get_by_id(cdr_id)['recordings'][0]
+        recording2 = self.call_logd.cdr.get_by_id(cdr_id)['recordings'][1]
+
+        self.call_logd.cdr.delete_recording_media(cdr_id, recording1['uuid'])
+
+        assert_that(
+            calling(self.call_logd.cdr.get_recording_media).with_args(
+                cdr_id, recording1['uuid']
+            ),
+            raises(CallLogdError).matching(
+                has_properties(status_code=400, error_id='recording-media-not-found')
+            ),
+        )
+        recordings = self.call_logd.cdr.get_by_id(cdr_id)['recordings']
+        assert_that(
+            recordings, has_item(has_entries(uuid=recording1['uuid'], deleted=True))
+        )
+
+        response = self.call_logd.cdr.get_recording_media(cdr_id, recording2['uuid'])
+        expected_filename = recording2['filename']
+        assert_that(response.text, equal_to('visible'))
+        assert_that(
+            response.headers['Content-Disposition'],
+            equal_to(f'attachment; filename={expected_filename}'),
+        )
+
+    @call_log(**{'id': 1}, recordings=[{'path': '/tmp/foobar.wav'}])
+    def test_delete_media_with_invalid_cdr(self):
+        rec_uuid = self.call_logd.cdr.get_by_id(1)['recordings'][0]['uuid']
+        assert_that(
+            calling(self.call_logd.cdr.delete_recording_media).with_args(2, rec_uuid),
+            raises(CallLogdError).matching(
+                has_properties(status_code=404, error_id='cdr-not-found-with-given-id')
+            ),
+        )
+
+    @call_log(**{'id': 1}, recordings=[{'path': '/tmp/foobar.wav'}])
+    @call_log(**{'id': 2}, recordings=[])
+    def test_delete_media_with_wrong_cdr(self):
+        rec_uuid = self.call_logd.cdr.get_by_id(1)['recordings'][0]['uuid']
+        assert_that(
+            calling(self.call_logd.cdr.delete_recording_media).with_args(2, rec_uuid),
+            raises(CallLogdError).matching(
+                has_properties(status_code=404, error_id='recording-not-found')
+            ),
+        )
+
+    @call_log(**{'id': 1}, recordings=[{'path': None}])
+    def test_delete_media_with_recording_already_deleted(self):
+        rec_uuid = self.call_logd.cdr.get_by_id(1)['recordings'][0]['uuid']
+        assert_that(
+            calling(self.call_logd.cdr.delete_recording_media).with_args(1, rec_uuid),
+            raises(CallLogdError).matching(
+                has_properties(status_code=400, error_id='recording-media-not-found')
+            ),
+        )
```

Comment:
I'm not sure if this behavior still make sense with your last modification?
It doesn't return error if your media is not found and deleted is false
But return an error if your media is not found and deleted is true
Maybe we should only return 204, deleted or not and doesn't bother the user with this logic. What do you think?